authority/provisioner/sign_ssh_options.go

```diff
@@ -411,25 +416,25 @@ type sshDefaultPublicKeyValidator struct{}
 // Valid checks that certificate request common name matches the one configured.
 func (v sshDefaultPublicKeyValidator) Valid(cert *ssh.Certificate, o SignSSHOptions) error {
 	if cert.Key == nil {
-		return errors.New("ssh certificate key cannot be nil")
+		return errs.BadRequest("ssh certificate key cannot be nil")
 	}
 	switch cert.Key.Type() {
 	case ssh.KeyAlgoRSA:
 		_, in, ok := sshParseString(cert.Key.Marshal())
 		if !ok {
-			return errors.New("ssh certificate key is invalid")
+			return errs.BadRequest("ssh certificate key is invalid")
```

Comment:
For consistency this should probably be 403

Reply:
Perhaps yes for consistency, but this is the only place where we check the key type, in a future PR we might change this validator to something else that checks the input, perhaps extending SSHCertOptionsValidator with the input key. This happens before the signing and should be a 400. For now, I will leave it as it is and add a comment.